### PR TITLE
Adding show by label feature

### PR DESCRIPTION
**Pull request by KharchenkoMaks**
Reworked show command, now asks user for label, if user enters a label, shows all tasks containing this label, if user enters nothing, shows all tasks with relations.

### CODE REVIEW

**Review by antonovchtd — APPROVED** [review]

File: src/view/commands/CommandResult.h [hunk 1]
```diff
@@ -11,18 +11,24 @@
 #include <optional>
 
 class CommandResult {
```
Comment (vladimir-moiseiev) [inline src/view/commands/CommandResult.h]: this class is a garbage can. it has everything. No good.

File: src/view/states/InputShowByLabelState.cpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+//
+
+#include "InputShowByLabelState.h"
+#include "view/user_interface/console_io/ConsoleUtilities.h"
+
+InputShowByLabelState::InputShowByLabelState(const std::shared_ptr<StatesFactory>& factory) : factory_(factory) {}
+
+std::shared_ptr<State> InputShowByLabelState::Execute(StateContext& context) {
+    const std::string users_label =
+            console_io::util::GetUserInput("Label to sort by", *factory_.lock()->GetConsolePrinter(), *factory_.lock()->GetConsoleReader());
```
Comment (vladimir-moiseiev) [inline src/view/states/InputShowByLabelState.cpp:12]: Please ask me about factory dependency here

**Review by vladimir-moiseiev — CHANGES_REQUESTED** [review]

File: src/model/TaskManager.cpp [hunk 3]
```diff
@@ -183,3 +183,23 @@ RelationalTask TaskManager::CreateTaskTransferFromTask(const std::map<TaskId, Ta
     }
     return task_transfer;
 }
+
+RelationalTask TaskManager::CreateRelationalTask(const TaskId& id, const TaskNode& task_node) {
```
Comment (IlliaPlaksa) [inline src/model/TaskManager.cpp:214]: Looks like it should be in refactoring PR

**Review by IlliaPlaksa — APPROVED** [review]

**Review by MichaelKupriianov — APPROVED** [review]

File: src/view/states/ShowState.cpp [hunk 4]
```diff
@@ -5,10 +5,10 @@
 #include "ShowState.h"
 #include "utilities/TaskConvertors.h"
 
-void ShowState::PrintTasks(const std::vector<RelationalTask> &tasks) {
-    for (auto task : tasks) {
+void ShowState::PrintTasks(const CommandResult::TasksToShow& tasks_to_show) {
+    for (auto task : tasks_to_show.tasks_) {
```
Comment (AntonApraksin) [inline src/view/states/ShowState.cpp]: Why iterating by value?

File: tests/view/StatesTests.cpp [hunk 5]
```diff
@@ -497,16 +498,42 @@ TEST_F(StatesTests, HelpStateExecute_ShouldPrintHelpMessage) {
 TEST_F(StatesTests, InputShowParametersStateExecute_ShouldCreateShowCommand) {
     // Arrange
     std::shared_ptr expected_next_state = std::make_shared<EndState>(nullptr);
+    std::shared_ptr<State> expected_state_machine_initial_state = std::make_shared<InputShowByLabelState>(nullptr);
+    std::unique_ptr<MockStateMachine> state_machine = std::make_unique<MockStateMachine>();
```
Comment (AntonApraksin) [inline tests/view/StatesTests.cpp:554]: You use `std::make_unique` here, but in `TaskManagerTest:576` you allocate with `new`. Be consistent and prefer `std::make_unique`.

**Review by AntonApraksin — CHANGES_REQUESTED** [review]

File: src/view/commands/CommandResult.h (hunk 1 above)
Comment (KharchenkoMaks) [reply src/view/commands/CommandResult.h]: Cleaned, left only result enum, TaskId, and vector of tasks in it.

File: src/view/states/ShowState.cpp (hunk 4 above)
Comment (KharchenkoMaks) [reply src/view/states/ShowState.cpp]: fixed

File: tests/view/StatesTests.cpp (hunk 5 above)
Comment (KharchenkoMaks) [reply tests/view/StatesTests.cpp:554]: fixed

**Review by AntonApraksin — APPROVED** [review]

**Review by vladimir-moiseiev — APPROVED** [review]

**Pull request merged**